### DISCUSSION TITLE
Current General Structure

### DISCUSSION THREAD

**Discussion by Utkar5hM** [NITK-KODE/cms-api#8] [General]
Current General structure for the backend

Type of Users

- Admins: Will basically handle the entire system.
- Mods: can Approve events created by event managers, can add Organizations and event managers to organizations.
- Eventmanagers: can handle Organizations, add members into organizations.
- users; just normal members who can join events participate can showcase awards into their profile

Now users will have points which they can earn by winning events, (haven't considered points by participating yet but can be done)

Now Routes
- /Organizations: from here u can view organizations.  their events, create events, handle events
- /organizations/organizationId/events : from here u can handle events , basically create events, etc.
- /organizations/organizationId/events/eventId/awards: for managing awards  . there will be set of templates given from which one can choose or go for the default ones.
- /events : for getting complete list of events. active(not completed) or completed events.
- /users : for managing personal users. 
- /users/username: for getting a profile page, where stuff will be showcased.

This kind of gives a fair idea. 
Removing the previous Mod event manager dashboard, now they can handle stuff from Organization board itself.

**Comment by Utkar5hM** [comment]
@anasaijaz  @BenzeneAlcohol 
Since you are working on frontend. :)

**Reply by anasaijaz** [reply]
We can use nested routes in express by passing the route object

Like 
Import userRouter from ...
router.use("/",userRouter)

And userRouter just exports the router object with get post etc methods

**Comment by Utkar5hM** [comment]
Since now we have added support for Institutes, the post is now outdated and a new discussion thread will be created soon with updated routes.
